flocker/control/httpapi.py

```diff
@@ -185,6 +185,143 @@ def saved(ignored):
         saving.addCallback(saved)
         return saving
 
+    @app.route("/datasets/<dataset_id>", methods=['POST'])
+    @user_documentation(
+        """
+        Update an existing dataset.
+        """,
+        examples=[
+            u"update dataset with primary",
+        ]
+    )
+    @structured(
+        # XXX Validation of the supplied primary address is performed by input
```

Comment:
At some point we will have some better concept of node identity, which will then allow validation. This is sufficient for now.